File: Delphi/SoapSecurity/getcert.pas
unit getcert;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls;

type
  TGetCertDialog = class(TForm)
    Label1: TLabel;
    Label2: TLabel;
    edtEmail: TEdit;
    edtStore: TEdit;
    btnOK: TButton;
    btnMessage: TButton;
    procedure FormCreate(Sender: TObject);
    procedure FormDestroy(Sender: TObject);
  private
  public
  end;

var
  GetCertDialog: TGetCertDialog;
  DefaultEmail: string = 'myname@mycompany.com';

implementation

{$R *.dfm}

procedure TGetCertDialog.FormCreate(Sender: TObject);
begin
  edtEmail.Text := DefaultEmail;
end;

procedure TGetCertDialog.FormDestroy(Sender: TObject);
begin
  DefaultEmail := edtEmail.Text;
end;

end.

File: CBuilder/EncryptDecrypt/getcert.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "getcert.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TGetCertDialog *GetCertDialog;

//---------------------------------------------------------------------------
__fastcall TGetCertDialog::TGetCertDialog(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TGetCertDialog::FormCreate(TObject *Sender)
{
  edtEmail->Text = DefaultEmail;
}
//---------------------------------------------------------------------------

void __fastcall TGetCertDialog::FormDestroy(TObject *Sender)
{
  DefaultEmail = edtEmail->Text;
}
//---------------------------------------------------------------------------
